### ISSUE TITLE
Follow cairo version numbering

### ISSUE THREAD

**Issue by julio4** [OpenZeppelin/cairo-contracts#781]
**🧐 Motivation**
From Scarb [v2.3.0-rc0](https://github.com/software-mansion/scarb/releases/tag/v2.3.0-rc0), Scarb vX.Y.Z matches Cairo vX.Y release. OpenZeppelin cairo contracts could follow the same standard for ease of use.

**📝 Details**
Starting from next version supporting Cairo `v2.3.0-rc0`

**Comment by martriay** [comment]
Why do you think this would be desirable?

**Comment by julio4** [comment]
Aren't you following the current scarb versioning? 
`openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", tag = "v0.7.0" }` for Scarb `0.7.0`

**Comment by martriay** [comment]
Yes it was a coincidence. We won't pair versioning with Scarb or Cairo, as Scarb did, because:

* they’re just not the same software/package, they shouldn’t behave as one
* they’re different codebases, they progress at different paces
* it’d be a huge coordination overhead and would result in slower release speed and inter-blockage
* it might confuse users! as it happened to you
* what if we need to introduce a breaking change? we’d be blocked by scarb and/or cairo adding breaking versions in their projects
* what if there’s a bug e.g. in Scarb, will it need to backport the fix to every version? Otherwise they'd be requiring their users to upgrade Cairo versions to get the fix (this happened to us).
* eventually i’d expect mismatching versions be compatible among Scarb, Cairo, and OZ Contracts
    * e.g. there’s a breaking change in the language but our contracts are still compatible with the previous and the new one
    * e.g. the same scarb installation allows you to compile with multiple cairo versions
        * maybe even at the same time! the important thing is that they’re all valid sierra

I frankly see little to no value and a bit of risk, as expressed above; but it’s ok if Scarb wants to experiment with this.
